salt/states/ssh_auth.py

```diff
@@ -308,14 +294,12 @@ def present(
     if source != "" and not source_path:
         data = "no key"
     elif source != "" and source_path:
-        key = __salt__["cp.get_file_str"](source, saltenv=__env__)
-        filehasoptions = False
-        # check if this is of form {options} {enc} {key} {comment}
-        sshre = re.compile(r"^(sk-)?(ssh\-|ecds).*")
-        key = key.rstrip().split("\n")
-        for keyline in key:
-            filehasoptions = sshre.match(keyline)
-            if not filehasoptions:
+        keys = __salt__["cp.get_file_str"](source, saltenv=__env__).rstrip().split("\n")
+        for key_line in keys:
+            key_data = _parse_line(key_line)
+            if "key" not in key_data:
+                continue
+            elif "options" not in key_data:
                 data = __salt__["ssh.set_auth_key_from_file"](
```

Comment:
This behaviour is unchanged and looks wrong to me.
It's passing the whole file for every key.
If e.g. the last line doesn't have options then it will all be overwritten.

Perhaps this whole branch is unnecessary?